### PR TITLE
Dev stage2 into DEV

### PR DESCRIPTION
**Pull request by zander-prinsloo**
1. Key values: removed as individual arguments, instead supplied as one list using argument `key_values` created using `gd_lq_key_values`
2. Remove unnecessary `dd` argument from `gd_compute_mld_*` and `gd_compute_watts_*`
3. This is potentially too much information, but here it is. The function `check_curve_validity_lq()` is called in `DEV` in the following ways. It is used a) line 115 in gd_compute_dist_stats.R in `gd_estimate_dist_stats_lq()`, where the r argument is squared as the input; b) line 784 in gd_compute_pip_stats_lq.R in `gd_estimate_lq()` where the r argument is not squared; c) in line 137 of gd_compute_poverty_stats.R in function `gd_estimate_poverty_stats_lq()` where the r argument is not squared; d) line 103  of prod_gd_compute_pip_stats_lq().R in function `prod_gd_estimate_lq()` where the r argument is not squared. Therefore, out of the four times it is called it is only the first where the r argument is squared. The paper (Villasenor & Arnold, 1989) uses the square root for the `r` argument, similar (although with a denominator) to `r` calculated in `gd_lq_key_values`. However, the other three calls of `check_curve_validity_lq()` mentioned above do not calculate the square root for r prior to the function call , which is why they just use the `r` argument directly. What has been done here is to i) use the square root for `r` in `key_values` list; ii) but then square the `r` argument within `check_curve_validity_lq()`. However, `r` is never actually used anywhere in `check_curve_validity_lq()` except in `if(r 0) {return(...)}` as an early check. This requires a bit more investigation to see whether the `r` argument is necessary here. 
4. `{collapse}` - use functions directly, and use `max()` and `sum()` rather than `fmax()` and `fsum()` because base functions are more efficient.
5. quantile functions in `md_compute_quantiles.R` that are efficient, utilize `lorenz` argument, and wrap around `md_compute_lorenz()`
6. Improvements in efficiency are made to `gd_compute_mld_lq()` and `gd_compute_quantile_lq()`. However, there are no changes in relation to the bug fixes yet. 
7. Removed the functions in md_quantile_functions.R. First ensured that a corresponding function that served the same purpose could be found in md_compute_quantiles.R

Some additional/general comments:
* Exported many additional functions.
* Some functions have been improved/vectorized. The new functions have taken the names of the old functions, and the old functions have the prefix "old_"

### CODE REVIEW

**Review by randrescastaneda — APPROVED** [review]
Hi @tonyfujs, 

All changes are approved. I added small modifications here and there that do not affect in any substantial sense the main contributions of the PR. 

All checks pass
image

**Comment by randrescastaneda** [comment]
Hi @tonyfujs , 

In this PR we have made several improvements to the efficiency of wbpip. In general, we are working towards vectorizing most of the functions, so that the API could also be vectorized, specially in the poverty line. To achieve that, we need to have the `wbpip` base working first. Could you please take a look and merge if ready? Thanks.  

image

**Review by tonyfujs — CHANGES_REQUESTED** [review]
Hi @zander-prinsloo - I'm getting an error when running `devtools::check()`
image

Can you please review? Thanks!

File: R/md_compute_poverty_stats.R [hunk 1]
```diff
@@ -157,9 +157,7 @@ md_compute_poverty_stats <- function(
 #'   md_compute_fgt(alpha = 2,
 #'                  return_data =  TRUE)
 #'
-#'
-#'
-#' dt_fgt <- data.table(povline = fgt$povline,
+#' dt_fgt <- data.frame(povline = fgt$povline,
```
Comment (zander-prinsloo) [inline R/md_compute_poverty_stats.R:160]: @tonyfujs I made the change, and re-documented. 

There is now no error when I run `devtools::check()` (see below) 

image

**Comment by tonyfujs** [comment]
Thanks @zander-prinsloo Everything looks good for me too now. The only other thing that I think we should do now, is remove all the old functionsas they are no longer needed

File: NAMESPACE [hunk 2]
```diff
@@ -11,21 +11,21 @@ export(derive_lq)
 export(fill_gaps)
 export(gd_clean_data)
 export(gd_compute_fit_lb)
-export(gd_compute_fit_lq)
```
Comment (shahronak47) [inline NAMESPACE]: Hi @zander-prinsloo , sorry to jump in but I am using `dev_stage2` for one of my task and I noticed that function `gd_compute_fit_lq` is no longer exported whereas `gd_compute_fit_lb` is exported which seems a bit odd to me. Is this intentional or has happened by mistake? 

cc : @randrescastaneda @tonyfujs

Comment (zander-prinsloo) [reply NAMESPACE]: @shahronak47 Thank you. The function `gd_compute_fit_lq()` has now been exported.

**Comment by zander-prinsloo** [comment]
> Thanks @zander-prinsloo Everything looks good for me too now. The only other thing that I think we should do now, is remove all the old functionsas they are no longer needed

@tonyfujs most recent commit removes all old functions with prefix "old_". The tests are also adjusted to account for their removal. CC @randrescastaneda 

All checks passing: 
image

**Comment by zander-prinsloo** [comment]
Last changes were to increase efficiency in `md_compute_polarization` and correct the tests. The benchmarks in the tests were slightly off (at around the fourth decimal). When using the formula to calculate the benchmarks analytically, the tests pass. 

Ready to review @tonyfujs @randrescastaneda 

All checks passing
image

**Review by randrescastaneda — APPROVED** [review]

**Comment by tonyfujs** [comment]
Great. Approved!

**Review by tonyfujs — APPROVED** [review]

**Pull request merged**